lib/bloc/fee_options/fee_options_bloc.dart

```diff
@@ -17,56 +17,78 @@ class FeeOptionsBloc extends Cubit<FeeOptionsState> {
   FeeOptionsBloc(this._breezSDK) : super(FeeOptionsState.initial());
 
   /// Fetches the current recommended fees
-  Future<List<FeeOption>> fetchFeeOptions(String toAddress) async {
+  Future<List<FeeOption>> fetchFeeOptions({required String toAddress, String? swapAddress}) async {
     RecommendedFees recommendedFees;
     try {
       recommendedFees = await _breezSDK.recommendedFees();
       _log.info(
         "fetchFeeOptions recommendedFees:\nfastestFee: ${recommendedFees.fastestFee},"
         "\nhalfHourFee: ${recommendedFees.halfHourFee},\nhourFee: ${recommendedFees.hourFee}.",
       );
-      return await _constructFeeOptionList(toAddress, recommendedFees);
+      return await _constructFeeOptionList(
+        toAddress: toAddress,
+        recommendedFees: recommendedFees,
+      );
     } catch (e) {
       _log.severe("fetchFeeOptions error", e);
       emit(FeeOptionsState(error: extractExceptionMessage(e, getSystemAppLocalizations())));
       rethrow;
     }
   }
 
-  Future<List<FeeOption>> _constructFeeOptionList(
-    String toAddress,
-    RecommendedFees recommendedFees,
-  ) async {
+  Future<List<FeeOption>> _constructFeeOptionList({
```

Comment:
Not entirely sure about the example code but could change into sth like this to reduce redundancy.
```
  Future<List<FeeOption>> _constructFeeOptionList(
    String toAddress,
    RecommendedFees recommendedFees,
  ) async {
    List<FeeOption> feeOptions = [
      for (var i = 0; i < 3; i++) await _generateFeeOption(i, toAddress, recommendedFees)
    ];
    emit(state.copyWith(feeOptions: feeOptions));
    return feeOptions;
  }

  Future<FeeOption> _generateFeeOption(
    int index,
    String toAddress,
    RecommendedFees recommendedFees,
  ) async {
    Set<int> waitingTimeSet = Set.unmodifiable({60, 30, 10});
    final Set<int> recommendedFeeSet = {
      recommendedFees.hourFee,
      recommendedFees.halfHourFee,
      recommendedFees.fastestFee,
    };
    final recommendedFee = recommendedFeeSet.elementAt(index);
    final fee = await _calculateTransactionFee(toAddress, recommendedFee);
    return FeeOption(
      processingSpeed: ProcessingSpeed.values.elementAt(index),
      waitingTime: Duration(minutes: waitingTimeSet.elementAt(index)),
      fee: fee,
      feeVByte: recommendedFee,
    );
  }
```

Reply:
I added this, but did not extract it as a function as I felt it was more readable. Let me know what you think